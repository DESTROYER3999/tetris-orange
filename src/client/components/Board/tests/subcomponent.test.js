import React from "react";
import { shallow, render } from "enzyme";
import toJson from "enzyme-to-json";
import BoardSub from "../subcomponent";

const board = [["", "", ""], ["", "", ""], ["", "", ""]];
describe("<BoardSub />", () => {
  test("Board rendering with a single board", () => {
    const wrapper = shallow(<BoardSub board={board} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("Board rendering with a spectre", () => {
    const wrapper = shallow(<BoardSub board={board} isSpectre={true} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("Board conditional css when isSpectre is true", () => {
    const wrapper = render(<BoardSub board={[]} isSpectre={true} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
