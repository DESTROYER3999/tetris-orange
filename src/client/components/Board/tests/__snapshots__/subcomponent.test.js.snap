// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BoardSub /> Board conditional css when isSpectre is true 1`] = `
<div
  class="sc-bdVaJa krbtIQ"
  id="board"
/>
`;

exports[`<BoardSub /> Board rendering with a single board 1`] = `
<styled.div
  id="board"
  isSpectre={false}
>
  <styled.div
    id="row"
    key="0"
  >
    <Connect(CellSub)
      color=""
      isSpectre={false}
      key="0"
    />
    <Connect(CellSub)
      color=""
      isSpectre={false}
      key="1"
    />
    <Connect(CellSub)
      color=""
      isSpectre={false}
      key="2"
    />
  </styled.div>
  <styled.div
    id="row"
    key="1"
  >
    <Connect(CellSub)
      color=""
      isSpectre={false}
      key="0"
    />
    <Connect(CellSub)
      color=""
      isSpectre={false}
      key="1"
    />
    <Connect(CellSub)
      color=""
      isSpectre={false}
      key="2"
    />
  </styled.div>
  <styled.div
    id="row"
    key="2"
  >
    <Connect(CellSub)
      color=""
      isSpectre={false}
      key="0"
    />
    <Connect(CellSub)
      color=""
      isSpectre={false}
      key="1"
    />
    <Connect(CellSub)
      color=""
      isSpectre={false}
      key="2"
    />
  </styled.div>
  <Connect(BoardOverlaySub) />
</styled.div>
`;

exports[`<BoardSub /> Board rendering with a spectre 1`] = `
<styled.div
  id="board"
  isSpectre={true}
>
  <styled.div
    id="row"
    key="0"
  >
    <Connect(CellSub)
      color=""
      isSpectre={true}
      key="0"
    />
    <Connect(CellSub)
      color=""
      isSpectre={true}
      key="1"
    />
    <Connect(CellSub)
      color=""
      isSpectre={true}
      key="2"
    />
  </styled.div>
  <styled.div
    id="row"
    key="1"
  >
    <Connect(CellSub)
      color=""
      isSpectre={true}
      key="0"
    />
    <Connect(CellSub)
      color=""
      isSpectre={true}
      key="1"
    />
    <Connect(CellSub)
      color=""
      isSpectre={true}
      key="2"
    />
  </styled.div>
  <styled.div
    id="row"
    key="2"
  >
    <Connect(CellSub)
      color=""
      isSpectre={true}
      key="0"
    />
    <Connect(CellSub)
      color=""
      isSpectre={true}
      key="1"
    />
    <Connect(CellSub)
      color=""
      isSpectre={true}
      key="2"
    />
  </styled.div>
</styled.div>
`;
